src/macros.jl

```diff
@@ -391,7 +391,7 @@ macro constraint(args...)
             lhs = :($(x.args[2]) - $(x.args[3]))
             newaff, parsecode = parseExprToplevel(lhs, :q)
             if vectorized
-                constraintcall = :(addconstraint.($m, constructconstraint!($newaff,$set)))
+                constraintcall = :(addconstraint.($m, constructconstraint!.($newaff,$set)))
```

Comment:
There's a subtle point here that the simple comparisons would never involve multidimensional sets, right?

Reply:
Simple comparisons meaning non-vectorized comparisons?

Reply:
Simple comparison in the sense of the comment above: ``# Simple comparison - move everything to the LHS``

Reply:
Yes, currently `@constraint m x .== y` returns a multidimensional array of the same shape of `x` and `y` with constraint references for the set `MOI.EqualTo`. We could instead return a single constraint with `MOI.Zeros` but then we need to transform `x-y` to a `Vector` first.
What do you prefer ?

Reply:
I prefer the current behavior, but it deserves some in-line comments.